### PR TITLE
Add support for getting password from file

### PR DESCRIPTION
**Pull request by kazan417**
Add support for getting password from file.
This is good feature for using docker secrets

### CODE REVIEW

**Comment by wxiaoguang** [comment]
Docker supports `env_file`: https://docs.docker.com/compose/how-tos/environment-variables/set-environment-variables/#use-the-env_file-attribute , https://docs.docker.com/compose/how-tos/environment-variables/variable-interpolation/#env-file

Does it work for you?

**Comment by wxiaoguang** [comment]
And Gitea has general "file to env" suppport in "environment-to-ini":

```
	Environment variables of the form "GITEA__SECTION_NAME__KEY_NAME__FILE"
	will be mapped to the ini section "[section_name]" and the key
	"KEY_NAME" with the value loaded from the specified file.
```

It is also packed into the docker image. So I think it doesn't need to add a new environment here?

File: docker/root/etc/s6/gitea/setup [hunk 1]
```diff
@@ -30,6 +30,9 @@ if [ ! -f ${GITEA_CUSTOM}/conf/app.ini ]; then
     if [ -n "$SECRET_KEY" ] && [ -z "$INSTALL_LOCK" ]; then
         INSTALL_LOCK=true
     fi
+    if  [  "$DB_PASSWD_FILE" ]; then
+        DB_PASSWD=`cat $DB_PASSWD_FILE`;
```
Comment (silverwind) [inline docker/root/etc/s6/gitea/setup:34]:
```suggestion
        DB_PASSWD="$(cat $DB_PASSWD_FILE)";
```

backtick syntax is legacy.

Comment (wxiaoguang) [reply docker/root/etc/s6/gitea/setup:34]: Before making more changes, I would suggest to use existing mechanisms. See my comments above.

Comment (silverwind) [reply docker/root/etc/s6/gitea/setup:34]: Yeah I guess there are better solutions than to introduce `_FILE` variables for every variable.

**Comment by kazan417** [comment]
> supports `env_file`: https://docs.docker.com/compose/how-tos/environment-variables/set-environment-variables/#use-the-env_file-attribute , https://docs.docker.com/compose/how-tos/environment-variables/variable-interpolation/#env-file
> 
> Does it work for

No this not work for swaem mode

**Comment by kazan417** [comment]
> And Gitea has general "file to env" suppport in "environment-to-ini" https://github.com/go-gitea/gitea/blob/main/contrib/environment-to-ini/environment-to-ini.go
> 
> ```
> 	Environment variables of the form "GITEA__SECTION_NAME__KEY_NAME__FILE"
> 	will be mapped to the ini section "[section_name]" and the key
> 	"KEY_NAME" with the value loaded from the specified file.
> ```
> 
> It is also packed into the docker image. So I think it doesn't need to add a new environment here?
> 
> https://github.com/go-gitea/gitea/blob/df9a78cd04e364264c103cf3a92d94179cc1dd4f/docker/root/etc/s6/gitea/setup#L60

Thats what i am looking for, but there is no such explanation in wiki. We need to add it into wiki

**Pull request closed without merging**